### PR TITLE
Add SRO conditional display

### LINKED ISSUE #1965: [SRO] Add simple conditional UI based on tags

### PR DESCRIPTION
**Pull request by nguyentvan7**
## Issue or discord link

- Resolve #1965

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [ ] I have commented my code in hard-to understand areas.
- [ ] I have made corresponding changes to README or wiki.
- [ ] For front-end changes, I have updated the corresponding English translations.
- [ ] I have run `yarn run mini-ci` locally to validate format and lint.
- [ ] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

File: libs/pando/ui-sheet/src/document.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+export interface FieldsDocument extends BaseDocument {
+  type: 'fields'
+  fields: Field[]
+}
+export interface ConditionalDocument extends BaseDocument {
+  type: 'conditional'
+  conditional: Conditional
+}
+
+export interface BaseDocument {
```
Comment (frzyc) [inline libs/pando/ui-sheet/src/document.ts:21]: would it make sense for a document to be componsed of multiple of these elements?

for example, if i want a 
```
HEADER
text
fields
conditional
```
all under one document header(like we have for talents)

File: libs/pando/ui-sheet/src/sheet.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import type { Document } from './document'
+
+export type UISheet = {
+  documents: Document[]
```
Comment (frzyc) [inline libs/pando/ui-sheet/src/sheet.ts:4]: would it make sense to associate each document with a key of some sort, for easier reverse lookup?

File: libs/pando/ui-sheet/src/document.ts (hunk 1 above)
Comment (nguyentvan7) [reply libs/pando/ui-sheet/src/document.ts:21]: image
This structure is a 1:1 match/parity of current ui sheet
blue = TextDocument
red = FieldsDocument
yellow = ConditionalDocument

green = optional header

So im not sure what you meant by the HEADER over multiple text/fields/conditional. I believe that is just created within the UI itself, not necessarily in the sheet

Comment (frzyc) [reply libs/pando/ui-sheet/src/document.ts:21]: Oh i misunderstand, I thought a character sheet is one UISheet, but i realized its a map of UIsheets, like
`{ auto:UISheets[], ...}`. I guess the naming convention confused me. Then disregard my comments, I am fine with the format for now.

Comment (nguyentvan7) [reply libs/pando/ui-sheet/src/document.ts:21]: A UI sheet is made up of an array or map of documents. I should adjust the shaping in sheet.ts prob

File: libs/pando/ui-sheet/src/sheet.ts (hunk 2 above)
Comment (nguyentvan7) [reply libs/pando/ui-sheet/src/sheet.ts:4]: Unresolving because yea we should map it, similar to current UI sheet structure

**Pull request closed without merging**